x-pack/test/security_solution_endpoint_api_int/apis/package.ts

```diff
@@ -0,0 +1,137 @@
+  Event,
+} from '../../../plugins/security_solution/common/endpoint/generate_data';
+import { FtrProviderContext } from '../ftr_provider_context';
+import { InsertedEvents, processEventsIndex } from '../services/resolver';
+
+interface EventIngested {
+  event: {
+    ingested: number;
+  };
+}
+
+interface NetworkEvent {
+  source: {
+    geo?: {
+      country_name: string;
+    };
+  };
+  destination: {
+    geo?: {
+      country_name: string;
+    };
+  };
+}
+
+const networkIndex = 'logs-endpoint.events.network-default';
+
+export default function ({ getService }: FtrProviderContext) {
+  const resolver = getService('resolverGenerator');
+  const es = getService('es');
+  const generator = new EndpointDocGenerator('data');
+
+  const searchForID = async <T>(id: string) => {
+    return es.search<SearchResponse<T>>({
+      index: eventsIndexPattern,
+      body: {
+        query: {
+          bool: {
+            filter: [
+              {
+                ids: {
+                  values: id,
+                },
+              },
+            ],
+          },
+        },
+      },
+    });
+  };
+
+  describe('Endpoint package', () => {
+    describe('ingested processor', () => {
+      let event: Event;
+      let genData: InsertedEvents;
+
+      before(async () => {
+        event = generator.generateEvent();
+        genData = await resolver.insertEvents([event]);
+      });
+
+      after(async () => {
+        await resolver.deleteData(genData);
+      });
+
+      it('sets the event.ingested field', async () => {
+        const resp = await searchForID<EventIngested>(genData.eventsInfo[0]._id);
+        expect(resp.body.hits.hits[0]._source.event.ingested).to.not.be(undefined);
+      });
+    });
+
+    describe('geoip processor', () => {
+      let processIndexData: InsertedEvents;
+      let networkIndexData: InsertedEvents;
+
+      before(async () => {
+        // 46.239.193.5 should be in Iceland
+        // 8.8.8.8 should be in the US
+        const eventWithBothIPs = generator.generateEvent({
+          extensions: { source: { ip: '8.8.8.8' }, destination: { ip: '46.239.193.5' } },
+        });
+
+        const eventWithSourceOnly = generator.generateEvent({
+          extensions: { source: { ip: '8.8.8.8' } },
+        });
+        networkIndexData = await resolver.insertEvents(
+          [eventWithBothIPs, eventWithSourceOnly],
+          networkIndex
+        );
+
+        processIndexData = await resolver.insertEvents([eventWithBothIPs], processEventsIndex);
+      });
+
+      after(async () => {
+        await resolver.deleteData(networkIndexData);
+        await resolver.deleteData(processIndexData);
+      });
+
+      it('sets the geoip fields', async () => {
+        const eventWithBothIPs = await searchForID<NetworkEvent>(
+          networkIndexData.eventsInfo[0]._id
+        );
+        expect(eventWithBothIPs.body.hits.hits[0]._source.source.geo?.country_name).to.be(
+          'United States'
```

Comment:
:grey_question: These always going to be in English, regardless of what locale it's running under?

Reply:
Yeah good point. I don't think they'd change much but for some reason the `8.8.8.8` ip address go switched to a different country this would fail. I'll switch it to check that the name isn't `undefined` 👍